File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionAtTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionAtTest extends BaseTest {
    @Test
    void inline() {
        var arr = fromJson("[4, 2, 13]");
        assertTransformation(arr, "$$at(0):$", fromJson("4"));
        assertTransformation(arr, "$$at(1):$", fromJson("2"));
        assertTransformation(arr, "$$at(-1):$", fromJson("13"));
        assertTransformation(arr, "$$at(3):$", null);
        assertTransformation(arr, "$$at:$", null);
    }

    @Test
    void object() {
        var arr = fromJson("[4, 2, 13]");

        assertTransformation(arr, fromJson("{ \"$$at\": \"$\", \"index\": 0 }"), fromJson("4"));
        assertTransformation(arr, fromJson("{ \"$$at\": \"$\", \"index\": 1 }"), fromJson("2"));
        assertTransformation(arr, fromJson("{ \"$$at\": \"$\", \"index\": -1 }"), fromJson("13"));
        assertTransformation(arr, fromJson("{ \"$$at\": \"$\", \"index\": 3 }"), null);
        assertTransformation(arr, fromJson("""
{
  "$$at": "$"
}
"""), null);
    }

}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionAt.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";
import { BigDecimal } from "../../functions/common/FunctionHelpers";

describe("TransformerFunctionAt", () => {
  test("inline", async () => {
    const arr = [4, 2, 13];
    await assertTransformation(arr, "$$at(0):$", 4);
    await assertTransformation(arr, "$$at(1):$", 2);
    await assertTransformation(arr, "$$at(-1):$", 13);
    await assertTransformation(arr, "$$at(3):$", null);
    await assertTransformation(arr, "$$at:$", null);
  });

  test("object", async () => {
    const arr = [4, 2, BigDecimal(13)];
    await assertTransformation(arr, { $$at: "$", index: 0 }, 4);
    await assertTransformation(arr, { $$at: "$", index: 1 }, 2);
    await assertTransformation(arr, { $$at: "$", index: -1 }, BigDecimal(13));
    await assertTransformation(arr, { $$at: "$", index: 3 }, null);
    await assertTransformation(arr, { $$at: "$" }, null);
  });
});
